src/orbital-helpers/orbital_embedding.cc

```diff
@@ -1,689 +1,772 @@
+                                    "larger than 0.5 are partitioned to B!",
+                                    i + frzopi[0], lo->get(0, i));
+                }
+            }
+        }
+        tmp = 0.0;
+        double cum_l_v = 0.0;
+        for (int i = 0; i < nrvirpi[0]; i++) {
+            tmp += lv->get(0, i);
+            cum_l_v = tmp / sum_lv;
+            if (cum_l_v < thresh) {
+                index_A_vir.push_back(i + offset_vec);
+            } else {
+                index_B_vir.push_back(i + offset_vec);
+                if (lv->get(0, i) > 0.5) {
+                    outfile->Printf("\n Warning! Virtual orbital %d have eigenvalue (%8.8f) "
+                                    "larger than 0.5 are partitioned to B!",
+                                    i + offset_vec, lv->get(0, i));
+                }
+            }
+        }
+    }
+
+    if (options.get_str("EMBEDDING_VIRTUAL_SPACE") == "PAO") {
+        outfile->Printf("\n ****** Build PAOs for virtual space ******");
+
+        // Call build_PAOs
+        double tau = options.get_double("PAO_THRESHOLD");
+        PAObuilder pao(Ca_save, frzopi + nroccpi + actv_a, ref_wfn->basisset());
+
+        bool fix_number = options.get_bool("PAO_FIX_VIRTUAL_NUMBER");
+
+        // ref_wfn->Ca()->print();
+        outfile->Printf("\n ****** Update C_vir ******");
+        // Write Ca
+        SharedMatrix C_pao = pao.build_A_virtual(nbf_A, tau);
+
+        int n_pao = 0;
+        if (!fix_number) {
+            n_pao = C_pao->ncol();
+        } else {
+            n_pao = index_A_vir.size();
+            ; // Fix virtual space to be equal to ASET partitioned virtual
+        }
+
+        Dimension nshortpi = nmopi;
+        nshortpi[0] = offset_vec + n_pao;
+        Dimension vir_start = nmopi;
+        vir_start[0] = offset_vec;
+
+        Slice dvir(vir_start, nshortpi);
+
+        if (!fix_number) {
+            ref_wfn->Ca()->set_block(mo, dvir, C_pao);
+        } else {
+            Dimension npao_fix = nmopi;
+            npao_fix[0] = n_pao;
+            Slice pao_fix(zeropi, npao_fix);
+            SharedMatrix C_pao_fix = C_pao->get_block(mo, pao_fix);
+            ref_wfn->Ca()->set_block(mo, dvir, C_pao_fix);
+        }
+
+        // B_vir ignored for now
+
+        outfile->Printf("\n ****** Create index lists ******");
+        // Form new index_A_vir and index_B_vir
+        index_A_vir.clear();
+        index_B_vir.clear();
+
+        // Number of virtual PAOs should be PAOs - active_virtual (?)
+        for (int i = 0; i < nrvirpi[0]; i++) {
+            if (i < n_pao) {
+                index_A_vir.push_back(i + offset_vec);
+                outfile->Printf("\n Orbital %d built as PAO", i + offset_vec + 1);
+            } else {
+                index_B_vir.push_back(i + offset_vec);
+            }
+        }
+
+        outfile->Printf("\n ****** PAOs done ******");
+    }
+

```

Comment:
After building PAOs, the code will run normally, with only virtual orbital changes to PAO